crates/ironrdp-pdu/README.md

```diff
@@ -400,3 +407,65 @@ MUST always be rejected. In such cases, an `enum` is deemed appropriate.
 [RangedU32_get]: https://docs.rs/deranged/0.3.8/deranged/struct.RangedU32.html#method.get
 [deranged]: https://docs.rs/deranged/0.3.8/
 [http-status-code]: https://docs.rs/http/0.2.9/http/status/struct.StatusCode.html
+
+## On bit flags
+
+Bit flags are used quite pervasively in the RDP protocol.
+IronRDP is relying on the [`bitflags` crate][bitflags] to generate well-defined flags structures,
+freeing us from worrying about bitwise logic.
+
+Three notable methods generated by the `bitflags` crate are:
+
+- [`from_bits`][from_bits],
+- [`from_bits_truncate`][from_bits_truncate], and
+- [`from_bits_retain`][from_bits_retain].
+
+Within IronRDP codebase, `from_bits_retain` should generally be used over `from_bits`, and
+`from_bits_truncate` is likely wrong. `from_bits_retain` will simply ignore unknown bits, but do
+not unset them (unlike `from_bits_truncate`), i.e.: the underlying `u32` is set exactly to the value
+received from the network.
+
+Rationale is:
+
+- PDU decoding and encoding logic should uphold the round-tripping property (`m = encode(decode(m))`),
+  and for this property to hold, parsing must be non-destructive (i.e.: lossless),
+  but `from_bits_truncate` is destructive (unknown bits are discarded).
+- Resilient parsing is generally preferred, ignoring unknown values as long as they are not needed and/or as
+  long as ignoring them is causing no harm, but `from_bits` is not lenient
+
+Note that it’s okay to use `from_bits` if strictness is actually required somewhere, but such places must be
+documented with a comment explaining why refusing unknown flags is better.
+
+`bitflags` v2.4 also introduced a new syntax in the `bitflags!` macro (<https://github.com/bitflags/bitflags/pull/371>):
+
+```rust
+bitflags! {
+    pub struct Flags: u32 {
+        const A = 0b00000001;
+        const B = 0b00000010;
+        const C = 0b00000100;
+
+        // The source may set any bits
+        const _ = !0; // <- This
+    }
+}
+```
+
+There is crate-level documentation for this: <https://docs.rs/bitflags/2.4.0/bitflags/#externally-defined-flags>
+
+This is making `from_bits_truncate` behaves exactly like `from_bits_retain` because all bits are
+then considered to be known and defined. In this case `from_bits` also never fails and provides no
+real benefit over `from_bits_retain`.
+
+Here is the bottom line:
+
+- Use **both** `from_bits_retain` and `const _ = !0` when resilient parsing is required.
+    - `const _ = !0` ensures we don’t accidentally have non resilient or destructive parsing.
```

Comment:
IMO this case is less compelling than the reason given by the crate itself (https://docs.rs/bitflags/2.4.0/bitflags/#externally-defined-flags)

Reply:
What do you mean? Are you referring to this?

> It helps compatibility when the external source may start setting additional bits at any time. 

This is what I mean by avoiding non resilient parsing. Maybe I should change the wording?

Reply:
I meant

> Generated methods like `all` and truncating operators like `!` only consider bits in defined flags. Adding an unnamed flag makes those methods consider additional bits, without generating additional constants for them.

is a more compelling reason to also add `const _ = !0` rather than protecting from accidentally having destructive parsing (e.g. if we use `from_bits` accidentally). Though arguably that all falls under the umbrella of your wording, so feel free to just ignore.